rfcs/proposed/task_group_wait_single_task/README.md

```diff
@@ -0,0 +1,489 @@
+# Waiting for Individual Tasks in ``task_group``
+
+This proposal introduces an extension for the oneTBB ``task_group`` and ``task_arena`` APIs, enabling the ability to wait for the completion of
+individual tasks.
+
+## Table Of Contents
+
+* 1 [Introduction](#introduction)
+* 2 [Proposal](#proposal)
+* 2.1 [Member functions of ``task_group`` class](#member-functions-of-task_group-class)
+* 2.2 [Member functions of ``task_arena`` class](#member-functions-of-task_arena-class)
+* 3 [Future Enhancements](#future-enhancements)
+* 3.1 [Run-and-wait methods for ``task_arena``](#run-and-wait-methods-for-task_arena)
+* 3.2 [Work Isolation Issue](#work-isolation-issue)
+* 4 [Exit Criteria and Open Questions](#exit-criteria-and-open-questions)
+* 5 [Implementation Details](#implementation-details)
+* 5.1 [Avoid Bypassing the Successor](#avoid-bypassing-the-successor)
+* 6 [Alternative Implementation Approaches](#alternative-implementation-approaches)
+* 6.1 [Assign Single ``wait_context`` to the Task](#assign-single-wait_context-to-the-task)
+* 6.2 [Avoid ``wait_context`` Usage](#avoid-wait_context-usage)
+
+## Introduction
+The ``tbb::task_group`` class represents a collection of tasks that execute concurrently. Tasks can be dynamically added to the group during execution.
+
+The [Task Group Dynamic Dependencies preview feature](../task_group_dynamic_dependencies/README.md) extends the ``task_group`` API to support task dependencies, ensuring that a *successor* task begins execution only after all its *predecessors* have completed.
+
+The current ``task_group`` API allows waiting for the completion of all tasks in the group using ``task_group::wait`` or ``task_group::run_and_wait``.
+The waiting thread may participate in executing other tasks within the ``task_arena``, which may not be related to the specific ``task_group``.
+
+Although waiting for all tasks suits many use cases, some workflows would benefit from the ability to wait for individual tasks.
+
+Consider an example using an out-of-order SYCL queue.
+
+```cpp
+sycl::queue q{sycl::property::queue::in_order{false}};
```

Comment:
I do not think the `in_order` property can be constructed as `false`; see https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#api:property-queue-in-order.